cairo/__init__.py

```diff
@@ -1,4 +1,24 @@
-from ._cairo import *  # noqa: F401,F403
+from pathlib import Path
+
+import os
+import sys
+
+if sys.platform == "win32" and sys.version_info >= (3, 8):
+    env_path = os.environ.get("PATH", "").split(os.pathsep)
+    first_gtk_dll_path = next(
+            filter(
+                lambda path: path is not None
+                and Path.is_file(Path(path) / "cairo.dll"),
+                env_path,
+            ),
+            None,
+    )
+    if first_gtk_dll_path:
+        with os.add_dll_directory(first_gtk_dll_path):
+            from ._cairo import *  # noqa: F401,F403
+
```

Comment:
If we were to do this, how about moving this code into a file called `cairo/_distributor_init.py`, following how numpy does it, https://github.com/numpy/numpy/blob/1d5b3397dc7beaf9ca7b2326127e74f55aecd162/numpy/_distributor_init.py? Also, I would prefer that file should by default be empty and only be added by the build system when appropriate, for example I don't want this to happen on the wheels distributed.

Reply:
Thanks! I'll have to think about possible options so that if someone gets the pycairo source code and runs `pip install .` or does `pip install --no-binary :all: pycairo` it gets the DLL code, but it doesn't with the statically built wheel.

Would you be open to the CI deleting some code during build?

Maybe we could detect that the .lib files exist during runtime, and if they don't, search for DLLs?

Reply:
We could also catch a DLL exception using a try-except, and only try to search for the DLLs if the user is about to get an error.

Reply:
Not sure. Anything should work but having it separated into a separate file like that is better. @lazka may have a better opinion.

Reply:
I ended up separating the DLL search logic in to a separate `windows_init` module. However, we will break the imports need to be at the module level rule if we move all the logic.